src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -432,9 +459,9 @@ public void close() throws Exception {
     isClosed = true;
     this.channelCache.closeAllChannels();
 
-    // Collect the perf metrics before closing if needed
-    if (metrics != null) {
-      Slf4jReporter.forRegistry(metrics).outputTo(logger.getLogger()).build().report();
+    // unregister jmx metrics
+    if (this.enableJmxMetrics && this.metrics != null) {
```

Comment:
nit: we should be checking `this.metrics` only

Reply:
if param is false, we wont have those metrics?